File: cpp/item.cpp
#include<string>
#include<iostream>
class item
{
private:
    int id;
    std::string name;
public:
    item(int id , std::string name){
        setId(id);
        setName(name);
    }

    int getId(){return id;}

    std::string getName(){return name;}

    void setName(std::string name){this->name = name;}

    void setId(int id){this->id=id;}

    std::string to_string();

    //operator < needed for creating set or this class as keys of map
    bool operator< (item const &i)const{
        return id < i.id;
    }
    
    ~item(){}
};

std::string item::to_string(){
    return "[Item(id="+std::to_string(this->id)+",name="+this->name+")]";
}

File: dart/Item.dart
class Item {
  late String _name;
  late int _id;
  late DateTime _manufactureTime;
  Item(String name, int id, DateTime manufactureTime) {
    _name = name;
    _id = id;
    _manufactureTime = manufactureTime;
  }
  //getters setters
  String get name => this._name;
  void set name(String n) {
    this._name = n;
  }

  int get id => this._id;
  void set id(int i) {
    this._id = i;
  }

  DateTime get manufactureTime => this._manufactureTime;
  void set manufactureTime(DateTime mft) {
    this._manufactureTime = mft;
  }

  @override
  String toString() {
    return 'Item(id=$_id,name=$name,manufactureTime=${_manufactureTime.toString()})';
  }
}
